src/GLua/AG/PrettyPrint.ag

```diff
@@ -519,18 +539,43 @@ sem Expr
                         loc.prettySingle = braces @lhs.ppconf @loc.emptyContents @fields.pretty
                         loc.emptyContents = toEmpty $ null @fields.copy
                         fields.indent = @lhs.indent + (if @fields.isMultiline then 1 else 0)
+                        fields.neighborPrecedence = 0
     | BinOpExpr         lhs.pretty = @left.pretty <-> @op.pretty <-> @right.pretty
+                        loc.precedence = case @op.copy of
+                                            AOr -> 1
+                                            AAnd -> 2
+                                            ALT -> 3
+                                            AGT -> 3
+                                            ALEQ -> 3
+                                            AGEQ -> 3
+                                            ANEq -> 3
+                                            AEq -> 3
+                                            AConcatenate -> 4
+                                            APlus -> 5
+                                            BinMinus -> 5
+                                            AMultiply -> 6
+                                            ADivide -> 6
+                                            AModulus -> 6
+                                            APower -> 8
+                        -- Note: the AST doesn't always represent precedence correctly in unparenthesized unary/binary expressions
```

Comment:
This sounds like a parser problem. Do you have an example where the AST represents precedence incorrectly?

Reply:
Yeah, there is indeed a parser problem. I tried to fix the parser but this solution was more within my haskell skill. The issue is basically that a mix of unary and binary operations aren't correct. IIRC, `-x + 1` puts the binary operation `x + 1` inside the unary operation so it parses more like `-(x + 1)`. Then, without my min-reduction of the precedence of unary/binary expressions, it would treat the expression as if it has precedence 7 instead of 5